commit 12a1e3786b3746339cc671054e2fcc99e9a0ef88
Author: Didier Raboud <odyx@debian.org>
Date:   Thu Jul 24 22:02:16 2014 +0200

    Correct default listening address used under systemd for new installations; also correct the discrepancy on upgrades
    
    Closes: #755807

diff --git a/debian/cups-daemon.preinst b/debian/cups-daemon.preinst
index 0d0a711..63087f0 100644
--- a/debian/cups-daemon.preinst
+++ b/debian/cups-daemon.preinst
@@ -10,10 +10,12 @@ install|upgrade)
         mv /etc/cups/cupsd.conf /etc/cups/cupsd.conf.conffile-bak
     fi
 
-    # If file doesn't exist or if it has two conflicting stanzas
+    # If file doesn't exist, has two conflicting stanzas or has a "listen-to-all" while cupsd.conf says it should 'listen-to-localhost'
     if [ ! -f /etc/cups/cupsd-systemd-listen.conf ] || \
        ( grep -q '^ListenStream=0.0.0.0:' /etc/cups/cupsd-systemd-listen.conf 2>/dev/null && \
-         grep -q '^ListenStream=127.0.0.1:' /etc/cups/cupsd-systemd-listen.conf 2>/dev/null ) ;\
+         grep -q '^ListenStream=127.0.0.1:' /etc/cups/cupsd-systemd-listen.conf 2>/dev/null ) ||
+       ( grep -q '^ListenStream=0.0.0.0:' /etc/cups/cupsd-systemd-listen.conf 2>/dev/null && \
+         grep -q '^\s*Listen localhost:' /etc/cups/cupsd.conf 2>/dev/null );\
     then
         mkdir -p /etc/cups
         cat >/etc/cups/cupsd-systemd-listen.conf <<EOF
@@ -43,8 +45,8 @@ EOF
         else
             cat >>/etc/cups/cupsd-systemd-listen.conf <<EOF
 # Matches the default 'Listen localhost:631' from cupsd.conf.default
-ListenStream=0.0.0.0:631
-ListenStream=[::]:631
+ListenStream=127.0.0.1:631
+ListenStream=[::1]:631
 EOF
         fi
     fi
